tsdb/db.go

```diff
@@ -645,6 +651,10 @@ func validateOpts(opts *Options, rngs []int64) (*Options, []int64) {
 	if opts.OutOfOrderTimeWindow < 0 {
 		opts.OutOfOrderTimeWindow = 0
 	}
+	walReplyConcurrency := runtime.GOMAXPROCS(0)
```

Comment:
I'm not sure that this logic should be here. In this PR, the logic to initialise the default is duplicated in two places (here and in the Head). Since the WAL replay is responsibility of the Head, I think we should just:
1. Leverage on `DefaultHeadOptions()` for the default value (see my other comment there)
2. Remove at all the logic here
3. Below, instead of doing `headOpts.WALReplayConcurrency = opts.WALReplayConcurrency`, I think you should just set `headOpts.WALReplayConcurrency` if `opts.WALReplayConcurrency > 0` (otherwise just use the default one)

Reply:
I agree with this. Default when this option is <=0, and use whatever the option is if it is >0. Maybe the user wants to experiment with more goroutines than there are processor cores :)